kanidmd/src/lib/idm/server.rs

```diff
@@ -39,10 +39,13 @@ use async_std::task;
 use core::task::{Context, Poll};
 use futures::task as futures_task;
 
-use concread::bptree::{BptreeMap, BptreeMapWriteTxn};
 use concread::hashmap::HashMap;
+use concread::{
+    bptree::{BptreeMap, BptreeMapWriteTxn},
+    CowCell,
+};
 use rand::prelude::*;
-use std::time::Duration;
+use std::{collections::BTreeSet, sync::Arc, time::Duration};
```

Comment:
I think this should be a hashset since this will be accessed very frequently with fast lookups :)

Reply:
Well then I suppose we should also change from BTreeSet to HashSet in authsession.rs too, right? 

Also, is there any easy way to convert from BTreeSet to HashSet? I am guessing I have to use a iterator and map from `Value` to `String` and insert the strings to a new HashSet.

Reply:
Yeah, when you are building the cache value during the fn commit, you'll take the btreeset of values from the entry, and then iterate and map that to a hashset<String> :)